Fix USDT approvals - using `safeApprove`

julianmrodri opened the pull request:
* Uses `safeApprove` instead of `safeIncreaseAllowance` to support the USDT case.
* Adapts both P1 and P0
*Adds a new integration test for regression testing

Resolves #383

tbrent commented:
Would be useful to add a regression test to the FacadeWrite tests that include USDT in both the Prime Basket and Backup Config. Otherwise implementation LGTM

julianmrodri commented:
Added the regression test. It cannot go in `FacadeWrite` though, because it is more an integration test where you need to use the real USDT contract. Added a new integration test that will always run in CI to confirm it works. It was failing before the change.

julianmrodri commented:
Merging based on offline positive feedback.
